core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -56,87 +58,31 @@ private V2Metadata() {
    * This is used to maintain compatibility with v2 by writing manifest list files with the old schema, instead of
    * writing a sequence number into metadata files in v2 tables.
    */
-  static class IndexedManifestFile implements ManifestFile, IndexedRecord {
+  static class IndexedManifestFile implements ManifestFile, Record {
```

Comment:
The Indexed* classes can still be used for writing. That way, we can convert the read path and then convert the write path later.

Reply:
Could you help to point out what do we need to do for write path?